RealmSwift/Sync.swift

```diff
@@ -253,7 +252,35 @@ public enum ClientResetMode {
     /// ```
     ///  For more details on the second block: ((Realm, Realm) -> Void)? = nil,
     /// - see: `RLMClientResetAfterBlock`
+    @available(*, deprecated, message: "Use discardUnsyncedChanges")
     case discardLocal(((Realm) -> Void)? = nil, ((Realm, Realm) -> Void)? = nil)
+    /// Identical in behavior to `.discardLocal.` Will fully replace
+    /// `.discardLocal.` once it's removed.
+    case discardUnsyncedChanges(((Realm) -> Void)? = nil, ((Realm, Realm) -> Void)? = nil)
```

Comment:
This will better to have its own dosctrings instead of linking to the Objective-C documentation, given that there is a callback which needs more detailed documentation